+++ b/src/backend/utils/cache/plancache.c
@@ -565,24 +565,22 @@ ReleaseGenericPlan(CachedPlanSource *plansource ADB_ONLY_COMMA_ARG(bool is_clust
                /* Drop this plan on remote nodes */
                if (is_cluster)
                {
+                       plansource->cluster_plan = NULL;
+               }else
+               {
+                       ListCell *lc;
+
                        /* Close any active planned Datanode statements */
-                       if (HaveActiveDatanodeStatements())
+                       foreach (lc, plan->stmt_list)
                        {
-                               ListCell *lc;
-                               foreach (lc, plan->stmt_list)
-                               {
-                                       Node *node = lfirst(lc);
+                               Node *node = lfirst(lc);

-                                       if (IsA(node, PlannedStmt))
-                                       {
-                                               PlannedStmt *ps = (PlannedStmt *)node;
-                                               drop_datanode_statements(ps->planTree, NULL, NULL);
-                                       }
+                               if (IsA(node, PlannedStmt))
+                               {
+                                       PlannedStmt *ps = (PlannedStmt *)node;
+                                       drop_datanode_statements(ps->planTree, NULL, NULL);
                                }
                        }
-                       plansource->cluster_plan = NULL;
-               }else
-               {
                        plansource->gplan = NULL;
                }
 #else